### PR TITLE
Implement channel archiving

### PR DESCRIPTION
**Pull request by cblgh**
This PR implements support for archiving of channels, see more background info in https://github.com/cabal-club/cabal-client/issues/68 and https://github.com/cabal-club/cabal-client/issues/16.

Archiving a channel is intended to hide the channel from channel browsers, and support for this will be added in cabal-client once this is finished & after this is merged. Only moderators or admins are allowed to archive channels—the local user is always allowed to archive channels, as they are always regarded as an admin in our subjective moderation system.

### CODE REVIEW

File: views/channel-archiving.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+          const pair = channel+key // a composite key, to know if we have seen this pair
+          if (msg.value.type === 'channel/archive') {
+            ops.push({
+              type: 'put',
+              key: `archive!${channel}!${key}`,
+              value: `${reason}`
+            })
+
+            if (!seen[pair]) events.emit('archive', channel, reason, key)
+            seen[pair] = true
```
Comment (noffle) [inline views/channel-archiving.js:65]: What is the logic around `seen` for?

**Review by noffle — COMMENTED** [review]
Nicely done @cblgh! This looks really good :)

File: views/channel-archiving.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+          if (!sanitize(msg)) { return }
+          const channel = msg.value.content.channel
+          const reason = msg.value.content.reason || ''
+          const key = msg.key
+          const pair = channel+key // a composite key, to know if we have seen this pair
+          if (msg.value.type === 'channel/archive') {
+            ops.push({
+              type: 'put',
+              key: `archive!${channel}!${key}`,
+              value: `${reason}`
```
[the remark this answers is not in the document]
Comment (cblgh) [reply views/channel-archiving.js]: fixed this! i saved the `key@seq` cause i figure that would save a concatenation step during lookup, lmk if that's silly and i'll just save the seq~

File: views/channel-archiving.js (hunk 1 above)
Comment (cblgh) [reply views/channel-archiving.js]: it wasn't really that well thoughtout, removed it–thanks!

**Comment by cblgh** [comment]
published as cabal-core@13.2.0 💖

**Pull request merged**